pkg/controller/pxcbackup/controller.go

```diff
@@ -850,17 +861,24 @@ func (r *ReconcilePerconaXtraDBClusterBackup) getBackupJob(
 	return job, nil
 }
 
-func (r *ReconcilePerconaXtraDBClusterBackup) cleanUpSuspendedJob(
+func (r *ReconcilePerconaXtraDBClusterBackup) cleanUpJob(
 	ctx context.Context,
 	cluster *api.PerconaXtraDBCluster,
 	cr *api.PerconaXtraDBClusterBackup,
+	preservePods bool,
 ) error {
 	job, err := r.getBackupJob(ctx, cluster, cr)
 	if err != nil {
 		return errors.Wrap(err, "get job")
 	}
 
-	if err := r.client.Delete(ctx, job); err != nil {
+	deleteOpts := []client.DeleteOption{}
+	if !preservePods {
+		// Job deletion preserves pods by default, we must set this option so that pods are deleted as well.
+		deleteOpts = append(deleteOpts, client.PropagationPolicy(metav1.DeletePropagationBackground))
+	}
```

Comment:
when job is suspended, pods are automatically terminated. i am not sure if we need this `preservePods` flag, we can set the propagation policy by default.